infrastructure/lambdaEdge.ts

```diff
@@ -2,41 +2,48 @@
     private args: LambdaEdgeArgs;
 
-    private callbackName: string;
-    private roleName: string;
-    private rolePolicyName: string;
-    private permissionName: string;
+    private readonly resourceNamesPrefix: string;
 
     private role: aws.iam.Role;
     private lambdaEdgeFunc: aws.lambda.CallbackFunction<any, any>;
 
-    constructor(name: string, callbackName: string, roleName: string, rolePolicyName: string, permissionName: string, args: LambdaEdgeArgs, opts: pulumi.ComponentResourceOptions) {
+    constructor(name: string, args: LambdaEdgeArgs, opts: pulumi.ComponentResourceOptions) {
         super("www-pulumi:infrastructure:LambdaEdge", name, undefined, opts);
 
-        // You might be wondering why these names need to be passed in when they could easily have
-        // been created dynamically based on the `name` of the component resource. The answer has to do
-        // with restrictions AWS places on deleting Lambda@Edge replicated functions (namely that they don't
-        // let you do it). At update-time, if Pulumi determines that a Lambda@Edge function needs to be
-        // deleted, (for example, when it's renamed), our create-before-delete functionality does the right thing,
-        // creating the replacement function and the CloudFront association, but fails on attempting to delete the
-        // previously associated function because of a bug in the AWS provider. So to safeguard against these
-        // failures, we opt to have the component accept these names explicitly.
-        // https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-edge-delete-replicas.html
+        const defaultRegion = aws.config.region;
+        if (defaultRegion !== aws.Region.USEast1 && !opts.providers && !opts.provider) {
+            throw new Error("Lambda@Edge must be deployed in us-east-1.");
+        }
 
-        this.callbackName = callbackName;
-        this.roleName = roleName;
-        this.rolePolicyName = rolePolicyName;
-        this.permissionName = permissionName;
+        // You might be wondering why the resources created in this component can't always have the prefix
+        // based on the `name` of the component resource. The answer has to do with restrictions AWS places
+        // on deleting Lambda@Edge replicated functions (namely that they don't let you do it).
+        // At update-time, if Pulumi determines that a Lambda@Edge function needs to be deleted,
+        // (for example, when it's renamed), our create-before-delete functionality does the right thing,
+        // creating the replacement function and the CloudFront association, but fails on attempting to delete
+        // the previously associated function because of a bug in the AWS provider. So to safeguard against
+        // these failures, we opt to have the component accept these names explicitly.
+        // https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/lambda-edge-delete-replicas.html
```

Comment:
Given the changes below (which seem fine), I'd probably cut this back and reword it to explain when you'd use `disable...` and when you wouldn't, and put the explanation on the interface property instead.